presentation/src/main/java/com/yapp/presentation/ui/admin/totalscore/AdminTotalScoreViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package com.yapp.presentation.ui.admin.totalscore
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.yapp.common.base.BaseViewModel
+import com.yapp.domain.usecases.GetAllMemberUseCase
+import com.yapp.presentation.model.Member
+import com.yapp.presentation.model.Member.Companion.mapTo
+import com.yapp.presentation.model.Team
+import com.yapp.presentation.model.collections.AttendanceList
+import com.yapp.presentation.ui.UPCOMING_SESSION_ID_KEY
+import com.yapp.presentation.ui.admin.totalscore.AdminTotalScoreContract.*
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AdminTotalScoreViewModel @Inject constructor(
+    private val getAllMemberUseCase: GetAllMemberUseCase,
+    private val savedStateHandle: SavedStateHandle,
+) : BaseViewModel<AdminTotalScoreUiState, AdminTotalScoreUiSideEffect, AdminTotalScoreUiEvent>(
+    AdminTotalScoreUiState()
+) {
+    init {
+        viewModelScope.launch {
+            setState { copy(loadState = AdminTotalScoreUiState.LoadState.Loading) }
+            getAllScores()
+        }
+    }
+
+    override suspend fun handleEvent(event: AdminTotalScoreUiEvent) {
+        when (event) {
+            is AdminTotalScoreUiEvent.OnBackArrowClick -> setEffect(AdminTotalScoreUiSideEffect.NavigateToPreviousScreen)
+        }
+    }
+
+    private suspend fun getAllScores() {
+        getAllMemberUseCase().collectWithCallback(
+            onSuccess = { memberEntity ->
+                val members = memberEntity.map { it.mapTo() }
+                val upcomingSessionId = savedStateHandle.get<Int>(UPCOMING_SESSION_ID_KEY)
+                    ?: AttendanceList.DEFAULT_UPCOMING_SESSION_ID
+                val groupMembers = members.groupBy {
+                    it.team
+                }
```

Comment:
```suggestion
                val upcomingSessionId = savedStateHandle.get<Int>(UPCOMING_SESSION_ID_KEY)
                    ?: AttendanceList.DEFAULT_UPCOMING_SESSION_ID
                val members = memberEntity.map { entity -> entity.mapTo() }
                                          .groupBy { member ->  member.team }

```

연산자를 체이닝 해도 좋을것 같아요!

Reply:
오 훨씬 깔끔하네요 감사합니다!